src/parser/parser.cpp

```diff
@@ -92,11 +93,59 @@ Expr Parser::getFunction(const std::string& name) {
   return getSymbol(name, SYM_VARIABLE);
 }
 
-Type Parser::getType(const std::string& var_name, SymbolType type) {
-  checkDeclaration(var_name, CHECK_DECLARED, type);
-  assert(isDeclared(var_name, type));
-  Type t = getSymbol(var_name, type).getType();
-  return t;
+Expr Parser::getVariableExpression(const std::string& name) {
+  Type t;
+  return getVariableExpressionForType(name, t);
+}
+
+Expr Parser::getVariableExpressionForType(const std::string& name, Type t) {
+  assert( isDeclared(name) );
+  // first check if the variable is declared and not overloaded
+  Expr expr = getVariable(name);
+  if(expr.isNull()) {
+    // the variable is overloaded, try with type if the type exists
+    if(!t.isNull()) {
+      // if we support annotations for function types, this will update to 
+      // separate t into ( argument types, return type )
```

Comment:
You're saying that if we later decide to support annotations, then...
right?
say "if we decide later..." to make it clear